cmd/kubeadm/app/util/config/cluster.go

```diff
@@ -39,54 +38,26 @@ import (
 	"k8s.io/kubernetes/cmd/kubeadm/app/constants"
 )
 
-// FetchConfigFromFileOrCluster fetches configuration required for upgrading your cluster from a file (which has precedence) or a ConfigMap in the cluster
-func FetchConfigFromFileOrCluster(client clientset.Interface, w io.Writer, logPrefix, cfgPath string, newControlPlane bool) (*kubeadmapi.InitConfiguration, error) {
-	// Load the configuration from a file or the cluster
-	initcfg, err := loadConfiguration(client, w, logPrefix, cfgPath, newControlPlane)
-	if err != nil {
-		return nil, err
-	}
-
-	// Apply dynamic defaults
-	if err := SetInitDynamicDefaults(initcfg); err != nil {
-		return nil, err
-	}
-	return initcfg, err
-}
-
-// loadConfiguration loads the configuration byte slice from either a file or the cluster ConfigMap
-func loadConfiguration(client clientset.Interface, w io.Writer, logPrefix, cfgPath string, newControlPlane bool) (*kubeadmapi.InitConfiguration, error) {
-	// The config file has the highest priority
-	if cfgPath != "" {
-		fmt.Fprintf(w, "[%s] Reading configuration options from a file: %s\n", logPrefix, cfgPath)
-		return loadInitConfigurationFromFile(cfgPath)
-	}
-
+// FetchConfigFromCluster fetches configuration from a ConfigMap in the cluster
+func FetchConfigFromCluster(client clientset.Interface, w io.Writer, logPrefix string, newControlPlane bool) (*kubeadmapi.InitConfiguration, error) {
```

Comment:
what about using `FetchInitConfigurationFromCluster` instead of `FetchConfigFromCluster` (same pattern of `LoadInitConfigurationFromFile` or `DefaultedInitConfiguration`)?

Reply:
Originally I envisioned `FetchConfigFromCluster` to be reduced to returning `ClusterConfiguration` only. However I am still not fully committed on this approach and this probably won't happen in 1.14, so I'll rename it to contain `Init` in its name.